docs/apps/components/rate.md

```diff
@@ -0,0 +1,168 @@
+<img src= "/apps/components/img/rate1.png">
+
+!!! Example
+    1. `this.addClassName(fields, className)`
+          * Go to Form Attribute :material-menu-right: Style Sheets :material-menu-right: add the class
+            ```
+            .abc{ // abc is the class name
+            background-color: red;
+            }
+            ```
+          * Follow the steps mentioned above, under Steps to use the methods for the Page Builder components
+          * ```
+            this.addClassName('rate_2n2vy8wj', 'abc')
+            ```
+
+    2. `this.removeClassName(fields, className)`
+        ```js
+        this.removeClassName('rate_2n2vy8wj', 'abc')
+        ```
+    
+    3. `this.getValues()`
+        ```js
+        var data = this.getValues();
+        console.log(data);
+        ```
+    
+    4. `this.hide(fields)`
+        ```js
+        var fields= ['rate_2n2vy8wj']
+        this.hide(fields)
+        ```
+    
+    5. `this.show(fields)`
+        ```js
+        var fields= ['rate_2n2vy8wj']
+        this.show(fields)
+        ```
+    
+    6. `this.disable(['fields'])`
+        ```js
+        this.disable(['rate_2n2vy8wj'])
+        ```
+    
+    7. `this.enable(['fields'])`
+        ```js
+        this.enable(['rate_2n2vy8wj'])
+        ```
+    
+    8.  `this.getValue('fieldName')`
+        ```js
+        var ratename = this.getValue('rate_2n2vy8wj');
+        console.log('getValue', ratename);
+        ```
+    
+    9. `this.getComponent('component_ID')`
+        ```js
+        var ratename = this.getComponent('rate_2n2vy8wj');
+        console.log('getComponent', ratename);
+        ```
+
+!!! info Add multiple classes
+    Use the following syntax to add multiple classes to a component:
+    ```js
+    this.addClassName('componentID', 'class1', 'class2')
+    ```
+
+## Config
+
+| **Name**|**Description**|**Icon**|
+|---------|---------------|--------|
+|**ID**| The rate component's unique identifier|<img src= "/apps/components/img/input_id.png">|
+|**Name**| [**Optional**] The display name of the rate field|<img src= "/apps/components/img/checkbox_name.png">|
+|**Width**| [**Optional**] Width of the field|<img src= "/apps/components/img/input_width.png">|
+|**Label Width**|Width of the label associated with an input field. It determines the horizontal space occupied by the label text|<img src= "/apps/components/img/input_labelwidth1.png">|
+|**Label Wrap**| If the label is longer than the allowed width the text will continue on another line|<img src= "/apps/components/img/input_labelwrap1.png">|
+|**Hide Label**| Hides the label on the form|<img src= "/apps/components/img/input_hidelabel.png">|
+|**Text Prompt**| A description to aid the user when completing the field|<img src= "/apps/components/img/input_textprompt.png">|
+|**Maximum**|Number of stars for rating|<img src= "/apps/components/img/rate_maximum.png">|
+|**Allow Half**|Enable this option to allow half rating|<img src= "/apps/components/img/rate_allowhalf.png">|
```

Comment:
Description should indicate that the half rating values are decimals like 3.5 .

Reply:
done